src/app/screens/do-routine/select-routine/index.tsx

```diff
@@ -19,52 +20,29 @@ import { Props } from './types';
 export const SelectRoutineScreen: FC<Props> = ({ navigation }) => {
   const { t } = useTranslation();
   const theme = useTheme();
+  const dispatch = useAppDispatch();
+  const { scheduleRoutines } = useAppSelector(state => state.routine);
 
   const routineOptions = useTranslatedOptions(weekDays, 'common:weekDay');
 
-  const [routineExercisesData, setRoutineExercisesData] = useState<
-    RoutineExercise[] | []
-  >([]);
-
-  const fetchRoutineExercisesData = (): Promise<RoutineExercise[]> => {
-    return new Promise((resolve, reject) => {
-      setTimeout(async () => {
-        try {
-          resolve(routineExercisesDataFile as RoutineExercise[]);
-        } catch (error) {
-          reject(error);
-        }
-      }, 1000);
-    });
-  };
-
   useEffect(() => {
-    const fetchData = async () => {
-      try {
-        const data = await fetchRoutineExercisesData();
-        setRoutineExercisesData(data);
-      } catch (error) {
-        console.error('Error fetching routines data:', error);
-      }
-    };
-
-    fetchData();
-  }, []);
+    dispatch(getMyScheduleRoutines());
+  }, [dispatch]);
 
   const { control, handleSubmit, watch } = useForm<FormData>({
     defaultValues: {
-      routine: '',
+      routine: scheduleRoutines[0]?.day,
     },
     resolver: yupResolver(validationSchema),
   });
 
   const selectedRoutine = watch('routine');
 
   const filteredExercises = useMemo(() => {
-    return routineExercisesData.filter(
-      exercise => exercise.routine === selectedRoutine,
-    );
-  }, [routineExercisesData, selectedRoutine]);
+    return scheduleRoutines
+      .filter(routine => routine.day === selectedRoutine)
```

Comment:
Why are you doing this filter?

Reply:
because im displaying the exercises based on the current routine selected

Reply:
Yes, im using due to the structure that we get on Routines: 
```
"exercises": [
                {
                    "id": "212428c6-fbe2-4c11-a800-93e5f84cf9ce",
                    "exerciseId": "1",
                    "name": "Remo con maquina",
                    "muscleGroup": "PECTORAL",
                    "repetitions": 5,
                    "restTimeSecs": 30,
                    "order": 1,
                    "series": [
                        {
                            "id": "39f2a6b9-c7ac-44fe-ade7-13e42f77a058",
                            "order": 1,
                            "weight": 80,
                            "weightMeasure": "kg",
                            "routineExerciseId": "212428c6-fbe2-4c11-a800-93e5f84cf9ce"
                        },
                ]
                
```
This is being used in the FlatList container: 
https://github.com/routine-fit/rosfit-mobile-app/blob/71982c2d294537a3e0d59da8647bddf7dca963a6/src/app/screens/do-routine/select-routine/styles.ts#L21

and in the renderExercise function: 
https://github.com/routine-fit/rosfit-mobile-app/blob/71982c2d294537a3e0d59da8647bddf7dca963a6/src/app/screens/do-routine/select-routine/index.tsx#L56